include/btllib/indexlr.hpp

```diff
@@ -22,6 +23,77 @@
 
 namespace btllib {
 
+struct Minimizer
+{
+  Minimizer() = default;
+
+  Minimizer(uint64_t min_hash,
+            uint64_t out_hash,
+            size_t pos,
+            bool forward,
+            std::string seq)
```

Comment:
I think having `std::string` as a argument here will trigger a string copy and affect performance, even though we're using move semantics in assignment (not sure but a copy of `seq` will be moved to `this->seq`?). Anyway, I'd use `std::string_view`, or just a const reference.

Reply:
If you have a `std::string` as a member, a copy needs to be made at some point. The `std::move` ensures it's only made once.

I'd recommend keeping `std::string` instead of `std::string_view` or reference, as it ensures the lifetime of the string is the same as the object. You can't guarantee the same with a reference or a view, and those are perhaps better used as temporary objects that are not stored.